src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,27 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASS_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws CustomException {
+        if (user.getLogin() == null) {
+            throw new CustomException("empty login");
+        }
+        if (user.getPassword() == null) {
+            throw new CustomException("empty password");
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new CustomException("the age is under 18");
+        }
+        if (user.getPassword().length() < MIN_PASS_LENGTH) {
+            throw new CustomException("password length must be more than 6");
```

Comment:
1. Use "MIN_PASSWORD_LENGTH" in message
2. Message is not correct. You say: "password length must be more than 6" It means that length 6 is not correct, but length 7 is correct